### PR TITLE
First version of extension template

### PR DESCRIPTION
**Pull request by russmiles**
Contributes to #1

### CODE REVIEW

File: chaosext/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+import os
+import os.path
+
+__version__ = '0.5.0'
```
Comment (Lawouach) [inline chaosext/__init__.py]: 0.1.0 :)

File: chaosext/actions.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+# -*- coding: utf-8 -*-
+import json
+import os.path
+from typing import Union
+
+from chaoslib.exceptions import FailedActivity
+from chaoslib.types import MicroservicesStatus, Secrets
```
Comment (Lawouach) [inline chaosext/actions.py]: drop the `MicroservicesStatus`. It'll be gone soon.

However, add `Configuration` instead please:

```
from chaoslib.types import Configuration, Secrets
```

File: chaosext/actions.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+# -*- coding: utf-8 -*-
+import json
+import os.path
+from typing import Union
```
Comment (Lawouach) [inline chaosext/actions.py]: You may drop those three imports

File: chaosext/probes.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+import json
+import os.path
+from typing import Union
+
+from chaoslib.exceptions import FailedActivity
+from chaoslib.types import MicroservicesStatus, Secrets
+import yaml
```
Comment (Lawouach) [inline chaosext/probes.py]: same as above and also remove the `import yaml`

File: pytest.ini [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+norecursedirs=dist build htmlcov docs .eggs
+addopts=-v -rxs --junitxml=junit-test-results.xml --cov=chaosk8s --cov-report term-missing:skip-covered --cov-report xml
```
Comment (Lawouach) [inline pytest.ini]: here, you should rename the project under coverage:

```
--cov=chaosext
```

The name is the name of the directory containing the source.

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+kubernetes
+chaostoolkit-lib>=0.6.0
+pycodestyle
+pyyaml
```
Comment (Lawouach) [inline requirements.txt]: remove that one please

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+kubernetes
```
Comment (Lawouach) [inline requirements.txt]: that one can go

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    'License :: OSI Approved :: Apache Software License',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: Implementation',
+    'Programming Language :: Python :: Implementation :: CPython'
+]
+author = 'chaostoolkit Team'
+author_email = 'contact@chaostoolkit.org'
```
Comment (Lawouach) [inline setup.py]: might be empty string?

File: setup.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    'Operating System :: OS Independent',
+    'License :: OSI Approved :: Apache Software License',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: Implementation',
+    'Programming Language :: Python :: Implementation :: CPython'
+]
+author = 'chaostoolkit Team'
```
Comment (Lawouach) [inline setup.py]: might be empty string?

File: setup.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+author_email = 'contact@chaostoolkit.org'
+url = 'http://chaostoolkit.org'
+license = 'Apache License Version 2.0'
+packages = [
+    'chaosext'
+]
+
+needs_pytest = set(['pytest', 'test']).intersection(sys.argv)
+pytest_runner = ['pytest_runner'] if needs_pytest else []
+test_require = [
```
Comment (Lawouach) [inline setup.py]: wonder if we shouldn't have a `requirements-dev.txt` instead?

File: README.md [hunk 11]
```diff
@@ -1,2 +1,36 @@
-# chaostoolkit-extension-template
-Template starting point for a new Python-based Chaos Toolkit extension
+# Chaos Toolkit Extension Template
```
Comment (Lawouach) [inline README.md]: should the README be targetting those who want to apply the template or the users of the resulting project?

File: chaosext/__init__.py (hunk 1 above)
Comment (russmiles) [reply chaosext/__init__.py]: Yes indeed!

File: chaosext/actions.py (hunk 2 above)
Comment (russmiles) [reply chaosext/actions.py]: Will do

File: requirements.txt (hunk 6 above)
Comment (russmiles) [reply requirements.txt]: Which one? Kubernetes is gone

File: setup.py (hunk 8 above)
Comment (russmiles) [reply setup.py]: No, I'd prefer this as a default for now. Obviously can be changed but just in case it isn't this can be a way we can offer support.

**Comment by Lawouach** [comment]
Generally looks like a good start. 

Perhaps we should update the readme to explain how to use this or use a generator like Atomist or cookicutter indeed.

**Comment by Lawouach** [comment]
Hey @russ, let's discuss about this PR tomorrow and merge it soon :)

**Comment by russ** [comment]
Hey guys. I would love to but you keep mentioning the wrong russ. 😜

**Comment by Lawouach** [comment]
Oh, damn, I apologise! I didn't verify what GH autoresolved to. I'm sorry for pinging you :(

**Comment by russ** [comment]
No worries. I always feel bad for others Russ’ that never get their
messages.

File: README.md (hunk 11 above)
Comment (russmiles) [reply README.md]: For now I'll leave as-is as I think this README content will migrate into custom-extension authoring documentation on chaostoolkit.org

File: setup.py (hunk 9 above)
Comment (russmiles) [reply setup.py]: With docs, this will be the next step.

File: setup.py (hunk 8 above)
Comment (russmiles) [reply setup.py]: With docs, this will be the next step.

File: setup.py (hunk 10 above)
Comment (russmiles) [reply setup.py]: Added.

**Pull request merged**